KomaMRICore/src/simulation/Functors.jl

```diff
@@ -52,9 +51,9 @@ See also [`f32`](@ref) and [`f64`](@ref) to change element type only.
 x = gpu(x, CUDABackend())
 ```
 """
-function gpu(x, backend::KA.GPU)
-    return fmap(x -> adapt(backend, x), x; exclude=_isleaf)
-end
+gpu(x, backend::KA.GPU) = fmap(x -> adapt(backend, x), x; exclude=_isleaf)
+adapt_storage(backend::KA.GPU, xs::MotionList) = MotionList(gpu.(xs.motions, Ref(backend)))
+adapt_storage(backend::KA.GPU, xs::Motion) = Motion(gpu(xs.action, backend), gpu(xs.time, backend), xs.spins)
```

Comment:
This lines should be here. All the Koma-related parts are at the end.

Reply:
Solved in [ee18d14](https://github.com/JuliaHealth/KomaMRI.jl/pull/472/commits/ee18d14dc9a047f7a87058bddf37ac8e537e8edb)